src/types.rs

```diff
@@ -314,46 +316,76 @@ where
         let offset = shape.indent + 1;
         Shape::legacy(budget, offset)
     };
+
     let list_lo = context.snippet_provider.span_after(span, "(");
-    let items = itemize_list(
-        context.snippet_provider,
-        inputs,
-        ")",
-        ",",
-        |arg| arg.span().lo(),
-        |arg| arg.span().hi(),
-        |arg| arg.rewrite(context, list_shape),
-        list_lo,
-        span.hi(),
-        false,
-    );
+    let (list_str, tactic) = if inputs.len() == 0 {
+        let tactic = if output.contains('\n') {
+            DefinitiveListTactic::Vertical
+        } else {
+            definitive_tactic(
+                &Vec::<ListItem>::new(),
+                ListTactic::HorizontalVertical,
+                Separator::Comma,
+                shape.width.saturating_sub(2 + output.len()),
```

Comment:
What does `2` represent ? Can you add a comment that describes its meaning ?
I've seen the following format: for example `// 1 = "{"` which means the digit 1 is about the opening curly bracket.